.github/actions/retest-action/entrypoint.sh

```diff
@@ -1,45 +1,69 @@
-#!/bin/sh
+#!/bin/bash
 
-set -ex
+set -euo pipefail
 
-if ! jq -e '.issue.pull_request' ${GITHUB_EVENT_PATH}; then
-    echo "Not a PR... Exiting."
-    exit 0
+# Check if this is a PR event
+
+if ! jq -e '.issue.pull_request' "${GITHUB_EVENT_PATH}"; then
+  echo "Not a PR... Exiting."
+  exit 0
 fi
 
-if [ "$(jq -r '.comment.body' ${GITHUB_EVENT_PATH})" != "/retest" ]; then
-    echo "Nothing to do... Exiting."
-    exit 0
+# Check if the comment is "/retest"
+
+if [ "$(jq -r '.comment.body' "${GITHUB_EVENT_PATH}")" != "/retest" ]; then
+  echo "Nothing to do... Exiting."
+  exit 0
 fi
 
-PR_URL=$(jq -r '.issue.pull_request.url' ${GITHUB_EVENT_PATH})
+# Get the PR details
 
-curl --request GET \
-    --url "${PR_URL}" \
-    --header "authorization: Bearer ${GITHUB_TOKEN}" \
-    --header "content-type: application/json" > pr.json
+PR_URL=$(jq -r '.issue.pull_request.url' "${GITHUB_EVENT_PATH}")
 
-ACTOR=$(jq -r '.user.login' pr.json)
-BRANCH=$(jq -r '.head.ref' pr.json)
+curl --silent --request GET \
+  --url "${PR_URL}" \
+  --header "authorization: Bearer ${GITHUB_TOKEN}" \
+  --header "content-type: application/json" > pr.json
 
-curl --request GET \
-    --url "https://api.github.com/repos/${GITHUB_REPOSITORY}/actions/runs?event=pull_request&actor=${ACTOR}&branch=${BRANCH}" \
-    --header "authorization: Bearer ${GITHUB_TOKEN}" \
-    --header "content-type: application/json" | jq '.workflow_runs | max_by(.run_number)' > run.json
+# Extract useful PR info
+
+PR_NUMBER=$(jq -r '.number' pr.json)
+PR_BRANCH=$(jq -r '.head.ref' pr.json)
+PR_COMMIT=$(jq -r '.head.sha' pr.json)
 
-RUN_URL=$(jq -r '.url' run.json)
+echo "Running /retest command for PR #${PR_NUMBER}"
+echo "PR branch: ${PR_BRANCH}"
+echo "Latest PR commit: ${PR_COMMIT}"
 
-curl --request POST \
+# Get failed Workflow runs for the latest PR commit
+
+curl --silent --request GET \
+  --url "https://api.github.com/repos/envoyproxy/envoy-mobile/actions/runs?branch=${PR_BRANCH}" \
+  --header "authorization: Bearer ${GITHUB_TOKEN}" \
+  --header "content-type: application/json" > runs.json
+jq ".workflow_runs[] | select(.head_sha == \"${PR_COMMIT}\") | select(.conclusion == \"failure\")" runs.json > failed_runs.json
```

Comment:
If needed we can expand this to more values beyond `failure`, especially `cancelled` and `timed_out`.

> Can be one of: completed, action_required, cancelled, failure, neutral, skipped, stale, success, timed_out, in_progress, queued, requested, waiting

https://docs.github.com/en/rest/actions/workflow-runs#get-a-workflow-run